pkg/v21/resource/configmapmigration/create.go

```diff
@@ -0,0 +1,76 @@
+package configmapmigration
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/pkg/v21/key"
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v21/key"
+	azurekey "github.com/giantswarm/cluster-operator/service/controller/azure/v21/key"
+	kvmkey "github.com/giantswarm/cluster-operator/service/controller/kvm/v21/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	objectMeta, err := r.getClusterObjectMetaFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Cluster is being deleted. No migration is necessary.
+	if key.IsDeleted(objectMeta) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cluster is being deleted")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil
+	}
+
+	chartSpecsToMigrate := r.newChartSpecsToMigrate()
+
+	if len(chartSpecsToMigrate) == 0 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no charts to migrate")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil
+	}
+
+	clusterConfig, err := r.getClusterConfigFunc(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Get all configmaps in the cluster namespace.
```

Comment:
For now we just get the configmaps in the cluster namespace in the CP.

Next PR will get the configmaps and chartconfig CRs in the tenant cluster which are also needed. The resource will be wired into the legacy controllers later.